File: JAVA/1545. Find Kth Bit in Nth Binary String.java
class Solution {
    public char findKthBit(int n, int k) {
        if(n==1)
        return '0';
        int l,m;
        l=(1<<n)-1;
        m=l/2+1;
        if(k==m)
        return '1';
        else if(k<m)
        return findKthBit(n-1,k);
        else
        return findKthBit(n-1,m-k+m)=='0'?'1':'0';
    }
}

File: C++/1545. Find Kth Bit in Nth Binary String.cpp
class Solution {
public:
    char findKthBit(int n, int k) {
        if(n == 1) return '0';
        int len = (1 << n) - 1;
        int mid = len / 2 + 1;
        if(k == mid) return '1';
        else if(k < mid) return findKthBit(n-1, k);
        else return findKthBit(n-1, mid - (k - mid)) == '0' ? '1' : '0';
    }
};
